### DISCUSSION TITLE
How to achieve a sliding expiration?

### DISCUSSION THREAD

**Discussion by ribtoks** [maypok86/otter#105] [Q&A] [answered]
Hi

Thank you for developing otter cache!

I'm mainly looking for it as a "local" Redis replacement. And the good thing with Redis is ability to extend expiration/TTL when a record is accessed (or even explicitly using EXPIRE). Maybe I just missed something obvious, but after looking into the implementation of `Get()`, I couldn't find any pointers.

Thank you

**Comment by maypok86** [comment]
Hi, unfortunately, this is not possible at the moment. I was planning to add various expiration strategies in v2, but I don't know when it will be released.

**Reply by ribtoks** [reply]
Thank you for your response. This is a bit unfortunate, but OK. I was using otter to keep user sessions and other data cached to prevent accessing main DB, but noticed that I really need to extend the TTL when I access it (especially this is relevant for "slower" underlying queries to main DB). Will be waiting when otter offers something like this to try it again!

**Reply by maypok86** [reply, marked as answer]
Hi, jfyi, [Otter v2](https://github.com/maypok86/otter/releases/tag/v2.0.0) ([godoc](https://pkg.go.dev/github.com/maypok86/otter/v2)) has been released and includes an `ExpiryAccessing` feature that might be helpful for your use case.

**Reply by ribtoks** [reply]
I discovered it literally yesterday night and currently integrating it! Thank you!
